src/editors/data/redux/app/reducer.js

```diff
@@ -40,15 +42,22 @@ const app = createSlice({
       blockValue: payload,
       blockTitle: payload.data.display_name,
     }),
-
     setStudioView: (state, { payload }) => ({ ...state, studioView: payload }),
     setBlockContent: (state, { payload }) => ({ ...state, blockContent: payload }),
     setBlockTitle: (state, { payload }) => ({ ...state, blockTitle: payload }),
     setSaveResponse: (state, { payload }) => ({ ...state, saveResponse: payload }),
     initializeEditor: (state) => ({ ...state, editorInitialized: true }),
-    setAssets: (state, { payload }) => ({ ...state, assets: payload }),
+    setImages: (state, { payload }) => ({
+      ...state,
+      images: { ...state.images, ...payload.images },
+      imageCount: payload.imageCount,
```

Comment:
Shouldn't this add to the existing imageCount? I'm assuming imageCount should always reflect number of `images`
```suggestion
      imageCount: state.imageCount + payload.imageCount,
```

Reply:
The image count is the total number of assets in a course, not the number in the most recent fetch. The image count should be overwritten each time to make sure the most recent asset total on each fetch.